server-spi/src/main/java/org/keycloak/models/ClientProvider.java

```diff
@@ -41,9 +57,23 @@
 
     RoleModel getClientRole(RealmModel realm, ClientModel client, String name);
 
-    Set<RoleModel> getClientRoles(RealmModel realm, ClientModel client);
+    @Deprecated
+    default Set<RoleModel> getClientRoles(RealmModel realm, ClientModel client) {
+        return getClientRolesStream(realm, client).collect(Collectors.toSet());
+    }
+
+    default Stream<RoleModel> getClientRolesStream(RealmModel realm, ClientModel client) {
+        return getClientRolesStream(realm, client);
```

Comment:
StackOverflowException

Reply:
It should be fixed already. Are you looking at latest change?